Mutagen.Bethesda.Analyzers.Engine/Config/Analyzer/AnalyzerConfig.cs

```diff
@@ -0,0 +1,37 @@
+﻿using Mutagen.Bethesda.Plugins;
+using Noggog;
+
+namespace Mutagen.Bethesda.Analyzers.Config.Analyzer;
+
+public interface IAnalyzerConfigLookup
+{
+    DirectoryPath? DataDirectoryPath { get; }
+    public bool LoadOrderSetByDataDirectory { get; }
+    public IEnumerable<ModKey>? LoadOrderSetToMods { get; }
+    GameRelease? GameRelease { get; }
+    public FilePath? OutputFilePath { get; }
+}
+
+public interface IAnalyzerConfig : IAnalyzerConfigLookup
+{
+    void OverrideDataDirectory(DirectoryPath path);
+    void OverrideLoadOrderSetByDataDirectory(bool value);
+    void OverrideGameRelease(GameRelease release);
+    void OverrideOutputFilePath(FilePath filePath);
+    void OverrideLoadOrderSetToMods(IEnumerable<ModKey> mods);
```

Comment:
Mentioned this before, but I think we'll eventually want to kill this API for the built in C# record init style API.   The tests do rely on this current setup as they use mocks + NSubstitutes `Received` style API, but I think those would just need a refactor a bit.   Simple enough to do
```cs
[Theory]
[AnalyzerInlineData("environment.data_directory = C:/some/path")]
[AnalyzerInlineData(@"environment.data_directory = C:\some\path")]
public void TestDataDirectory(
    string line,
    IAnalyzerConfig config,
    AnalyzerConfigReader sut)
{
    sut.ReadInto(line.AsSpan(), config);
    config.Received(1).OverrideDataDirectory("C:/some/path");
}
```
to just do
```cs
[Theory]
[AnalyzerInlineData("environment.data_directory = C:/some/path")]
[AnalyzerInlineData(@"environment.data_directory = C:\some\path")]
public void TestDataDirectory(
    string line,
    AnalyzerConfig config,
    AnalyzerConfigReader sut)
{
    sut.ReadInto(line.AsSpan(), config);
    config.DataDirectoryPath
        .Should().Be(new DirectoryPath("C:/some/path"));
}
```
Which, id argue is better anyway to not rely on mocks (I went through a lonnng testing journey.. coming out the other side as a "mock minimally" guy)

-----------

Don't necessarily have to do these changes in this PR.  If you dont want, I can just crack open a card for later

Reply:
Yeah I also prefer that! Didn't want to change the established way tests were written in this project though haha. I can change it when we have figured out the rest here :)